parser/parser.go

```diff
@@ -46,76 +50,54 @@ type ConfigDoc struct {
 	Filepath   string
 }
 
-// ConfigManager the implementation of ReadUnmarshaller and io.Reader
-// byte storage.
-type ConfigManager struct {
-	parser         Parser
-	configContents map[string][]byte
-}
-
 // BulkUnmarshal iterates through the given cached io.Readers and
 // runs the requested parser on the data.
-func (s *ConfigManager) BulkUnmarshal(configList []ConfigDoc) (map[string]interface{}, error) {
-	if err := s.setConfigs(configList); err != nil {
-		return nil, fmt.Errorf("set configuration: %w", err)
-	}
-
-	var allContents = make(map[string]interface{})
-	for filepath, config := range s.configContents {
-		var singleContent interface{}
-		if err := s.parser.Unmarshal(config, &singleContent); err != nil {
-			return nil, fmt.Errorf("parser unmarshal: %w", err)
-		}
-
-		allContents[filepath] = singleContent
-	}
-
-	return allContents, nil
-}
-
-func (s *ConfigManager) setConfigs(configList []ConfigDoc) error {
-	s.configContents = make(map[string][]byte)
+func BulkUnmarshal(configList []ConfigDoc, input string) (map[string]interface{}, error) {
+	configContents := make(map[string][]byte)
 	for _, config := range configList {
 		contents, err := ioutil.ReadAll(config.ReadCloser)
-		defer config.ReadCloser.Close()
```

Comment:
Because we're in a loop and there's potential for _many_ files to be opened, using `defer` means the file will remain opened until the end of the function call, in this case `BulkUnmarshal`. So I removed the `defer` to make sure we're closing immediately after we're done with it.

For the close on error bit, this should be a non-issue as an error will terminate `conftest` and the OS will handle any of the required closings. We don't need to manually close them before returning the error.